### PR TITLE
Custom signin

### PR DESCRIPTION
**Pull request by yramanchuk**
implement custom sign-in controller

### CODE REVIEW

File: FirebaseAuthUI/FIRPasswordSignInViewController.h [hunk 1]
```diff
@@ -24,31 +24,59 @@ NS_ASSUME_NONNULL_BEGIN
     @brief The view controller that asks for user's password.
  */
 @interface FIRPasswordSignInViewController : FIRAuthUIBaseViewController
+{
+  @protected
+  /** @var _email
+   @brief The @c email address of the user from the previous screen.
+   */
+  NSString *_email;
```
Comment (morganchen12) [inline FirebaseAuthUI/FIRPasswordSignInViewController.h]: Why not use a property?

File: samples/objc/FirebaseUI-demo-objc/Samples/Auth/FIRAuthViewController.m [hunk 2]
```diff
@@ -81,11 +81,13 @@ - (void)viewWillAppear:(BOOL)animated {
...
 
+  [self.navigationController setToolbarHidden:NO animated:YES];
 }
 
--(void)viewDidDisappear:(BOOL)animated {
-  [super viewDidDisappear:animated];
+-(void)viewWillDisappear:(BOOL)animated {
+  [super viewWillDisappear:animated];
   [self.auth removeAuthStateDidChangeListener:self.authStateDidChangeHandle];
+  [self.navigationController setToolbarHidden:YES animated:YES];
```
Comment (morganchen12) [inline samples/objc/FirebaseUI-demo-objc/Samples/Auth/FIRAuthViewController.m]: You should probably pass the `animated` flag through to `setToolbarHidden:animated:`.

```objc
[self.navigationController setToolbarHidden:YES animated:animated];
```

That way your code doesn't create animations if they're not going to be visible anyway.

File: FirebaseAuthUI/FIRPasswordSignInViewController.h (hunk 1 above)
Comment (yramanchuk) [inline FirebaseAuthUI/FIRPasswordSignInViewController.h]: I don't want to expose this property to any other classes; only parent classes should have access to it.

File: samples/objc/FirebaseUI-demo-objc/Samples/Auth/FIRAuthViewController.m (hunk 2 above)
Comment (yramanchuk) [inline samples/objc/FirebaseUI-demo-objc/Samples/Auth/FIRAuthViewController.m]: fixed, thanks

**Pull request merged**